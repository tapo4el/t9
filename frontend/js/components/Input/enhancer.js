import { connect } from 'react-redux';

// const mapStateToProps = state => ({
//     text: state.predictionMask,
// });
const mapStateToProps = state => {
    console.log(state);
    return {
        text: state.predictionMask.join(''),
    }
};

export default connect(mapStateToProps, null);
