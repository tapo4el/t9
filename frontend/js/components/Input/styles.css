.t9-app input.textInput {
    width: 500px;
    line-height: 40px;
    padding: 0 10px;
    outline: 0;
    border-radius: 10px;
    border: 1px solid #e0e1e2;
    font-size: 14px;
}
