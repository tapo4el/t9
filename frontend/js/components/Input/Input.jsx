import React from 'react';
import { string } from 'prop-types';

import './styles.css';

class Input extends React.PureComponent {
    static propTypes = {
        inputText: string.isRequired,
    };

   render() {
       const { inputText } = this.props;
       return (
           <input type="text" className="textInput" value={inputText} readOnly/>
       );
   }
};

export default Input;
