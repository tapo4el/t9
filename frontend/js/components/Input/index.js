import Input from './Input';
import enhancer from './enhancer';

export default enhancer(Input);
