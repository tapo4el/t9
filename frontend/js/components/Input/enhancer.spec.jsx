import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Input from './index';

describe('Input enhancer', () => {
    const initialState = {
        inputText: ['Hello'],
        predictionMask: [{symbol: 'a', number: '2'}]
    };
    const mockStore = configureStore();

    it ('should render correctly', () => {
        const component = renderer.create(
            <Input store={mockStore(initialState)}/>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});

// describe('Input enhancer', () => {
//     it ('should rendered correct', () => {
//
//     });
// });
