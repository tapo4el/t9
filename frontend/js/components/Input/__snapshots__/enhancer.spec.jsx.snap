// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input enhancer should render correctly 1`] = `
<input
  className="textInput"
  readOnly={true}
  type="text"
  value="Hello a"
/>
`;
